### ISSUE TITLE
Make AI Model Training work with tasks

### ISSUE THREAD

**Issue by daniel-wer** [scalableminds/webknossos#8265]
## Detailed Description
As far as I can see two things are missing:
- [x] Allow to specify task IDs instead of annotation IDs as the training data source. To do this I think it would be easiest to look up the (or are there multiple?) annotation ID for a specific task ID and continue to use that. This way, no changes in the worker should be needed. (@fm3 Could you comment on whether that's possible?)
- [x] Take into account task bounding boxes and not only user bounding boxes

## Context

* [x] Specific to long-running jobs (set `jobsEnabled=true` in `application.conf`)

Training data for model trainings is often created using tasks. It should be possible to start a training from these.

**Comment by fm3** [comment]
Sounds good! Yes, there may be multiple annotation ids per task. We should also decide if only “finished” instances should be used, or also “active” ones.

**Comment by MichaelBuessemeyer** [comment]
>  We should also decide if only “finished” instances should be used, or also “active” ones.

Maybe add a warning for each unfinished task? This way we can allow both and thus this feature is more flexible :thinking:

**Comment by daniel-wer** [comment]
I would vote to only use finished instances (and show a warning that there are x unfinished instances that are not used). It usually does not make sense to use unfinished training data.

**Comment by MichaelBuessemeyer** [comment]
> This way, no changes in the worker should be needed.

Sadly, the worker needs adjustments as the bounding boxes collected by the frontend are not passed to the job and thus the worker collect the bounding boxes itself.

**Comment by daniel-wer** [comment]
> Sadly, the worker needs adjustments as the bounding boxes collected by the frontend are not passed to the job and thus the worker collect the bounding boxes itself.

But the worker previously collected the bounding boxes as well. Or is the needed change that the `taskBoundingBox` in addition to `userBoundingBoxes` are collected by the worker?

**Comment by MichaelBuessemeyer** [comment]
> Or is the needed change that the taskBoundingBox in addition to userBoundingBoxes are collected by the worker?

Yes, that what I meant. Sorry for being unclear here :see_no_evil:
